### PR TITLE
Move extension/storage to extension/experimental/storage

### PR DESCRIPTION
**Pull request by JamesJHPark**
**Description:** 
This PR is to move the experimental package `extension/storage` to `extension/experimental/storage` and follow the extension packages naming conventions. 

**Link to tracking Issue:** <Issue number if applicable>
#4059

### CODE REVIEW

**Review by codeboten — APPROVED** [review]
Change looks good here, please make sure to update the contrib repo as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ## 🛑 Breaking changes 🛑
 
+- Move extension/storage to extension/experimental/storageextension (#4082).
```
Comment (Aneurysm9) [inline CHANGELOG.md:7]:
```suggestion
- Move `extension/storage` to `extension/experimental/storageextension` (#4082).
```

**Review by Aneurysm9 — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@codeboten @Aneurysm9 see comment in the issue, do we follow the same pattern for just interfaces?

**Comment by Aneurysm9** [comment]
> @codeboten @Aneurysm9 see comment in the issue, do we follow the same pattern for just interfaces?

Which pattern? Having `extension` at the end of the name or having `experimental` in the path?  I think they're probably both fine, but I could be convinced that the suffix might not be necessary.

**Comment by bogdandrutu** [comment]
Was referring to the suffix since it is not an extension itself

**Comment by JamesJHPark** [comment]
@bogdandrutu storageextension suffix has been removed to storage. Thank you.

**Review by bogdandrutu — APPROVED** [review]

**Comment by alolita** [comment]
Hi @bogdandrutu can you please do a final review and merge? ty!

**Comment by bogdandrutu** [comment]
@JamesJHPark please make sure that the contrib is fixed if needed.

**Pull request merged**